scipy/meson.build

```diff
@@ -223,7 +222,16 @@ cython_gen_cpp = generator(cython_cli,
   output : '@BASENAME@.cpp',
   depends : [_cython_tree])
 
-c_undefined_ok = ['-Wno-maybe-uninitialized']
+# Check if compiler flags are supported. The standard convention used here is
+# to drop the leading dash, and turn remaining dashes into underscores
+Wno_maybe_uninitialized = []
+if cc.has_argument('-Wno-maybe-uninitialized')
+  Wno_maybe_uninitialized += ['-Wno-maybe-uninitialized']
+endif
```

Comment:
@eli-schwartz does this look like the right thing to do (also style-wise), or is there a more idiomatic way?

Reply:
Perhaps instead use:

```meson
Wno_maybe_uninitialized = cc.get_supported_arguments('-Wno-maybe-uninitialized')
Wno_discarded_qualifiers = cc.get_supported_arguments('-Wno-discarded-qualifiers')
```

:)